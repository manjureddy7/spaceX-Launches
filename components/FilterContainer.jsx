import React from "react";
import YearContainer from "./Filters/YearContainer";
import LaunchContainer from "./Filters/LaunchContainer";
import LandingContainer from "./Filters/LandingContainer";
import styles from "../styles/FilterContainer.module.css";

const FilterContainer = () => {
  return (
    <div className={styles.RootContainer} data-testid="FilterContainer">
      <h2>Filters</h2>
      <YearContainer />
      <LaunchContainer />
      <LandingContainer />
    </div>
  );
};

export default FilterContainer;
