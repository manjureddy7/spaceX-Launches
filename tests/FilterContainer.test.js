import { render, cleanup } from "@testing-library/react";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import FilterContainer from "../components/FilterContainer";
import router from "../helper/helper";

afterEach(cleanup);

describe("FilterContainer Component", () => {
  const { getByTestId, asFragment } = render(
    <RouterContext.Provider value={router}>
      <FilterContainer />
    </RouterContext.Provider>
  );
  test("Should render correctly on route", async () => {
    expect(getByTestId("FilterContainer")).toBeInTheDocument();
  });
  test("Should match snapshot", () => {
    expect(asFragment(<FilterContainer />)).toMatchSnapshot();
  });
});
