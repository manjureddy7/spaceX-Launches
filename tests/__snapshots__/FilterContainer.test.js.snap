// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterContainer Component Should match snapshot 1`] = `<DocumentFragment />`;
